modules/migrations/gitea.go

```diff
@@ -59,6 +62,7 @@ func NewGiteaLocalUploader(ctx context.Context, doer *models.User, repoOwner, re
 		repoName:    repoName,
 		prHeadCache: make(map[string]struct{}),
 		userMap:     make(map[int64]int64),
+		prCache:     make(map[int64]*models.PullRequest),
```

Comment:
Will this maintain all migrated pull requests in memory? If that's the case, it will possibly also hold all of their associated issues, owner, poster, etc. Can't this be a problem for migrating large repositories? libravatar implements a limited-size last-use cache that can be used here. It wouldn't even need to be thread-safe. (Note: I see that this was being done for issues and users as well).